urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/tools/PropTools.java

```diff
@@ -0,0 +1,115 @@
+package terminal.tools;
+
+import java.io.*;
+import java.util.Properties;
+
+public class PropTools {
+
+	 /**
+     * Load a Properties File
+     * @param propsFile
+     * @return Properties
+     * @throws java.io.IOException
+     */
+    public static Properties load(String propsFile)
+    {
+    	Properties props = new Properties();
+        FileInputStream fis;
+		try {
+			fis = new FileInputStream(new File(propsFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fis*. The resource is referenced at line 20. The resource is closed at line 21. There are other execution paths that do not contain closure statements, for example, when Properties.load throws an exception. Ensure *fis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).